### DISCUSSION TITLE
Need help querying data from 2 different locations

### DISCUSSION THREAD

**Discussion by RobbeBuyle** [tinacms/tinacms#3170] [Help] [answered]
Hi all!

I am struggling trying to query data from post-collection and display it on the index page (where all my blocks are showed). My goal is to show the 4 most recent blog posts on the home page but I'm having a hard time making it work.
For the schema I used roughly the same format as the starter, I'm just having problems defining the 'postData' and querying it.

Thanks in advance!

`export default function Home(props) {`

`const { data } = useTina({
    query: props.query,
    variables: props.variables,
    data: props.data,
  });`

`return (...`
`{data && data.page.blocks
      ? data.page.blocks.map(function (block, i) {
        switch (block.__typename) {
          case ... `

`export const getStaticProps = async () => {
  const { data, query, variables } = await client.queries.page({
    relativePath: "home.mdx",
  });`

  `return {
    props: {
      data,
      query,
      variables,
    },
  };
};`

**Comment by RobbeBuyle** [comment, marked as answer]
UPDATE: correct me if I'm wrong but I think I am supposed to extend my graphQL query.
I followed the docs but it is still unclear to me how I would go about and integrate this.
https://tina.io/docs/data-fetching/custom-queries/

**Reply by jamespohalloran** [reply]
That's right! Our "Tina Cloud Starter" has an example of this, where they one query that included the postConnection, and the layout:
https://github.com/tinacms/tina-cloud-starter/blob/main/.tina/queries/queries.gql#L1

**Reply by RobbeBuyle** [reply]
Thanks so much for taking the time to reply to this. I managed to fix it by declaring the proper gql variables like last and sort and it worked from then on out.
